packages/plugin/src/rules/naming-convention.ts

```diff
@@ -225,61 +283,67 @@ const rule: GraphQLESLintRule<NamingConventionRuleConfig> = {
       ObjectTypeDefinition: node => {
         if (options.ObjectTypeDefinition) {
           const property = normalisePropertyOption(options.ObjectTypeDefinition);
-          checkNode(node.name, property.style, 'Type', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Type');
         }
       },
       InterfaceTypeDefinition: node => {
         if (options.InterfaceTypeDefinition) {
           const property = normalisePropertyOption(options.InterfaceTypeDefinition);
-          checkNode(node.name, property.style, 'Interface', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Interface');
         }
       },
       EnumTypeDefinition: node => {
         if (options.EnumTypeDefinition) {
           const property = normalisePropertyOption(options.EnumTypeDefinition);
-          checkNode(node.name, property.style, 'Enumerator', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Enumerator');
         }
       },
       InputObjectTypeDefinition: node => {
         if (options.InputObjectTypeDefinition) {
           const property = normalisePropertyOption(options.InputObjectTypeDefinition);
-          checkNode(node.name, property.style, 'Input type', property.suffix, property.prefix);
+          checkNode(node.name, property, 'Input type');
         }
       },
-      FieldDefinition: node => {
-        if (options.FieldDefinition) {
+      FieldDefinition: (node: any) => {
```

Comment:
I think the `node` type in  [typekit](https://github.com/dotansimha/graphql-eslint/blob/9899c0b1e824cdda9835b22893459853e1242cf9/packages/plugin/src/testkit.ts#L9) could be updated to contain `parent`:

```
export type GraphQLESlintRuleListener<WithTypeInfo extends boolean> = {
  [K in keyof ASTKindToNode]?: (
    node: GraphQLESTreeNode<ASTKindToNode[K], WithTypeInfo> & {
      parent: GraphQLESTreeNode<ASTKindToNode[K], WithTypeInfo>;
    }
  ) => void;
} &
  Record<string, any>;
```

However, I don't know what effect it has in other places so I'll let you decide whether I should update it :)